### PR TITLE
Py38 feature prep

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Update core requirements to 3.8 compatible versions.
- Cut py38 wheels
- Move docker images over to using python 3.8

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -216,8 +216,8 @@ commands:
       - store_artifacts:
           path: test-reports
 
-  deploy_to_pypi:
-    description: "Publish packages to pypi"
+  deploy_pypi_prolouge:
+    description: "Check docker smoke test results"
```
Comment (vEpiphyte) [inline .circleci/config.yml:220]: words

File: .circleci/config.yml [hunk 2]
```diff
@@ -596,6 +632,16 @@ workflows:
             branches:
               ignore: /.*/
 
+      - deploy_pypi_37:
+          requires:
+            - doctests
+            - python37
```
Comment (vEpiphyte) [inline .circleci/config.yml:638]: py38

Comment (vEpiphyte) [reply .circleci/config.yml:638]: replays

File: setup.py [hunk 3]
```diff
@@ -49,12 +49,12 @@ def run(self):
...
+        'xxhash>=1.4.4,<2.0.0',
         'lmdb>=1.0.0,<2.0.0',
         'tornado>=6.0.3,<7.0.0',
         'regex>=2020.5.14',
-        'PyYAML>=5.1,<6.0',
-        'aiohttp>=3.5.4,<4.0',
+        'PyYAML>=5.2,<6.0',
+        'aiohttp>=3.6.3,<4.0',
         'prompt-toolkit>=3.0.4,<3.1.0',
         'lark-parser>=0.9.0,<0.10.0',
```
Comment (jnwatson) [inline setup.py:59]: Please to update to `lark-parser>=0.9.0,<0.11.0`

File: .circleci/config.yml [hunk 4]
```diff
@@ -216,8 +216,8 @@ commands:
       - store_artifacts:
           path: test-reports
 
-  deploy_to_pypi:
-    description: "Publish packages to pypi"
+  deploy_pypi_prolouge:
+    description: "Common Pypi prolouge"
```
Comment (jnwatson) [inline .circleci/config.yml:220]: speeling "prologue"

**Review by jnwatson — CHANGES_REQUESTED** [review]

**Pull request merged**